lib/bango/views/premium.py

```diff
@@ -0,0 +1,46 @@
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+
+from lib.bango.constants import BANGO_ALREADY_PREMIUM_ENABLED
+from lib.bango.errors import (
+    BangoError, BangoFormError, BangoImmediateError, ProcessError)
+from lib.bango.forms import MakePremiumForm
+from lib.bango.serializers import SellerProductBangoOnly
+from lib.bango.views.base import BangoResource
+
+
+@api_view(['POST'])
+def premium(request):
+    view = BangoResource()
+    view.error_lookup = {
+        'INVALID_COUNTRYISO': 'currencyIso',
+    }
+
+    try:
+        serial, form = view.process(
+            serial_class=SellerProductBangoOnly,
+            form_class=MakePremiumForm,
+            request=request)
+    except ProcessError, exc:
+        return exc.response
+
+    data = form.cleaned_data
+    data['bango'] = serial.object['seller_product_bango'].bango_id
+    if not data['bango']:
+        # Note: that this error formatting seems quite inconsistent
+        # with the rest of the errors. Something we should clean updatecache(
+        raise BangoImmediateError(
+            {'seller_product_bango':
+             ['Empty bango_id for: {0}'
+              .format(serial.object['seller_product_bango'].pk)]}
+        )
+
+    try:
+        view.client('MakePremiumPerAccess', data,
+                    raise_on=[BANGO_ALREADY_PREMIUM_ENABLED])
+    except BangoFormError, exc:
+        return Response(status=204)
```

Comment:
I see what's happening now. I don't think you should rename BangoFormError because that's what it is. This code is pretty confusing though. I'd suggest adding a comment on this line like "This is one of our expected exceptions as defined above."